src/simplicial-hott/08-covariant.rzk.md

```diff
@@ -814,8 +814,82 @@ is covariant as shown above. Transport of an `e : C x` along an arrow
     ( center-contraction (dhom-from A x y f C u) (is-covariant-C x y f u))
     ( lift)
     ( homotopy-contraction (dhom-from A x y f C u) (is-covariant-C x y f u) lift)
+
+#def covariant-uniqueness-curried
+  ( A : U)
+  ( x y : A)
+  ( f : hom A x y)
+  ( C : A → U)
+  ( is-covariant-C : is-covariant A C)
+  ( u : C x)
+  : ( v : C y)
+  -> ( dhom A x y f C u v)
+  -> ( covariant-transport A x y f C is-covariant-C u) = v
+  :=
+    \ v g -> covariant-uniqueness A x y f C is-covariant-C u (v, g)
+
 ```
 
+We show that for each `v : C y`, the  map `covariant-uniqueness` is an equivalence.
+This follows from the fact that the total spaces (summed over `v : C y`)
+of both sides are contractible.
+
+```rzk title="RS17, Lemma 8.15"
+
+#def total-map-covariant-uniqueness-curried-is-equiv
+  ( A : U)
+  ( x y : A)
+  ( f : hom A x y)
+  ( C : A -> U)
+  ( is-covariant-C : is-covariant A C)
+  ( u : C x)
+  : is-equiv
+      (Sigma (v : C y), dhom A x y f C u v)
```

Comment:
Here, `Sigma` should be replaced with `Σ`. I've realised we do not have this listed in the styleguide https://rzk-lang.github.io/sHoTT/STYLEGUIDE/#use-of-unicode-characters.

Reply:
thanks! I'll do that as soon I figure out how to input unicode in my editor